content/client-lib-development-guide/features.textile

```diff
@@ -488,6 +490,11 @@ h3(#realtime-channel). Channel
 ** @(RTL4f)@ Once an @ATTACH@ @ProtocolMessage@ is sent, if an @ATTACHED@ @ProtocolMessage@ is not received within the "default realtime request timeout":#defaults, the attach request should be treated as though it has failed and the channel should transition to the @SUSPENDED@ state. The channel will then be subsequently automatically re-attached as described in "RTL13":#RTL13
 ** @(RTL4d)@ If the language permits, a callback can be provided that is called when the channel is attached successfully or the attach fails and the @ErrorInfo@ error is passed as an argument to the callback
 ** @(RTL4e)@ If the user does not have sufficient permissions to attach to the channel, the channel will transition to @FAILED@ and set the @Channel#errorReason@
+** @(RTL4j)@ If the attach is not a clean attach (where a 'clean attach' is an attach attempt where the channel has either not previously been attached or has been explicitly detached since the last time it was attached); for example an automatic reattach triggered by "@RTN15c3@":#RTN15c3 or "@RTL13a@":#RTL13a (non-exhaustive); the library should set the "@ATTACH_RESUME@":#TR3f flag in the @ATTACH@ message
+** @(RTL4k)@ If the user has specified a non-empty @params@ object in the @ChannelOptions@ ("@TB2c@":#TB2c), it must be included in a @params@ field of the @ATTACH@ @ProtocolMessage@
+*** @(RTL4k1)@ If @params@ are sent in the @ATTACH@ message, the corresponding @ATTACHED@ (and any subsequent @ATTACHED@s) will contain the subset of those params that the server recognises and validates. This should be exposed in a read-only @params@ field of the @Channel@ (or a @getParams()@ method where that is more idiomatic)
+** @(RTL4l)@ If the user has specified a @modes@ array in the @ChannelOptions@ ("@TB2d@":#TB2d), it must be encoded as a bitfield per "@TR3@":#TR3 and set as the @flags@ field of the @ATTACH@ @ProtocolMessage@. (For the avoidance of doubt, when multiple different spec items require flags to be set in the @ATTACH@, the final @flags@ field should be the bitwise OR of them all)
+** @(RTL4m)@ On receipt of an @ATTACHED@, the client library should decode the @flags@ into an array of @ChannelMode@s (that is, the same format as @ChannelOptions.modes@) and expose it as a read-only @modes@ field of the @Channel@ (or a @getModes()@ method where that is more idiomatic). This should only contain @ChannelMode@s: it should not contain flags which are not modes (see "@TB2d@":#TB2d)
```

Comment:
P.S. I don't think `@ChannelMode@s` renders correctly in Textile

Reply:
Where in other libs do we do this `get` variation on an attribute such as `getModes`?

Reply:
my impression was that ably-java does that when it wants to enforce read-only access, eg https://github.com/ably/ably-java/blob/b3bd021154f8da6d0389683ba60788d19c31b7a9/lib/src/main/java/io/ably/lib/rest/Auth.java#L916